src/Agent.Listener/Configuration/InteractiveSessionConfigurationManager.cs

```diff
@@ -0,0 +1,213 @@
+        {
+            regHelper.SetRegistry(WellKnownRegistries.ScreenSaver, "0");
+            regHelper.SetRegistry(WellKnownRegistries.ScreenSaverDomainPolicy, "0");
+
+            ShowAutoLogonWarningIfAlreadyEnabled(regHelper, userName);
+
+            var windowsHelper = HostContext.GetService<INativeWindowsServiceHelper>();
+            windowsHelper.SetAutoLogonPassword(logonPassword);
+
+            regHelper.SetRegistry(WellKnownRegistries.AutoLogonUserName, userName);
+            regHelper.SetRegistry(WellKnownRegistries.AutoLogonDomainName, domainName);
+            regHelper.SetRegistry(WellKnownRegistries.AutoLogon, "1");
+
+            var startupProcessPath = Path.Combine(HostContext.GetDirectory(WellKnownDirectory.Bin), "agent.service.exe");
+            var startupCommand = string.Format("{0} {1}", startupProcessPath, "runAsProcess");
+            regHelper.SetRegistry(WellKnownRegistries.StartupProcess, startupCommand);
+
+            regHelper.SetRegistry(WellKnownRegistries.ShutdownReason, "0");
+            regHelper.SetRegistry(WellKnownRegistries.ShutdownReasonUI, "0");
+
+            regHelper.SetRegistry(WellKnownRegistries.LegalNoticeCaption, "");
+            regHelper.SetRegistry(WellKnownRegistries.LegalNoticeText, "");
+
+            var processInvoker = HostContext.GetService<IProcessInvoker>();
+            processInvoker.ExecuteAsync(
+                            workingDirectory: string.Empty,
+                            fileName: "powercfg.exe",
+                            arguments: "/Change monitor-timeout-ac 0",
+                            environment: null,
+                            cancellationToken: CancellationToken.None).Wait();
+
+            processInvoker.ExecuteAsync(
+                            workingDirectory: string.Empty,
+                            fileName: "powercfg.exe",
+                            arguments: "/Change monitor-timeout-dc 0",
+                            environment: null,
+                            cancellationToken: CancellationToken.None).Wait();
+        }
+
+        public bool RestartNeeded()
+        {
+            var regHelper = new WindowsRegistryHelper(HostContext.GetService<IWindowsRegistryManager>());
+            var userName = regHelper.GetRegistry(WellKnownRegistries.AutoLogonUserName);
+            var domainName = regHelper.GetRegistry(WellKnownRegistries.AutoLogonDomainName);
+
+            if(string.IsNullOrEmpty(userName) || string.IsNullOrEmpty(domainName))
+            {
+                return false;
+            }
+            var nativeWindowsHelper = HostContext.GetService<INativeWindowsServiceHelper>();
+            return !nativeWindowsHelper.IsTheSameUserLoggedIn(domainName, userName);
+        }
+
+        public void UnConfigure()
+        {
+            throw new NotImplementedException();
```

Comment:
what's the path for customer if he enabled auto-logon and now wants to disable or come out of interactive session. where are we removing what we setup. I expect the unconfigure should bring back machine to original state